pennylane/workflow/get_best_diff_method.py

```diff
@@ -63,20 +55,20 @@ def handle_return(transform):
     @wraps(qnode)
     def wrapper(*args, **kwargs):
         device = qnode.device
-        tape = qml.workflow.construct_tape(qnode)(*args, **kwargs)
 
-        config = _make_execution_config(None, "best")
+        # Construct the tape using the same method as the execution workflow
+        tape = qml.workflow.construct_tape(qnode, level="user")(*args, **kwargs)
```

Comment:
```suggestion
        batch = qml.workflow.construct_batch(qnode, level="user")(*args, **kwargs)
```
should allow batch transforms too.